nixos/modules/services/networking/firewall.nix

```diff
@@ -493,6 +508,22 @@ in
         '';
     };
 
+    networking.firewall.extraAllowed = mkOption {
+      type = types.listOf (types.submodule { options = {
+           source = mkOption { type = types.string; default = ""; example = "example.com"; description = "Source of the packets"; };
+           protocol = mkOption { type = types.enum ["tcp" "udp" "icmp"]; default = ""; example = "tcp"; description = "Protocol used."; };
+           dport = mkOption { type = types.int; default = 0; example = 22; description = "Destination port of the packet."; };
+           useIPv4 = mkOption { type = types.bool; default = true; description = "Use the IPv4 address of this host."; };
+           useIPv6 = mkOption { type = types.bool; default = true; description = "Use the IPv6 address of this host."; };
```

Comment:
In fact it's unclear what "host" mean. Should that be "from this source"? Can source be a CIDR prefix?